x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/MvExpandOperator.java

```diff
@@ -107,8 +108,8 @@ public final Page getOutput() {
                 noops++;
                 Page result = prev;
                 prev = null;
-                expandingBlock = null;
-                expandedBlock = null;
+
+                releaseAndClearState();
```

Comment:
With the changes in `Block.expand()`, I think we can remove `mayHaveMultivaluedFields == false` section to simplify this method.